update/process/shardBlock.go

```diff
@@ -171,6 +190,135 @@ func (s *shardBlockCreator) createMiniBlockHeaders(body *block.Body) (int, []blo
 	return totalTxCount, miniBlockHeaders, nil
 }
 
+func (s *shardBlockCreator) saveAllTransactionsToStorageIfSelfShard(
+	shardHdr *block.Header,
+	body *block.Body,
+) {
+	if shardHdr.GetShardID() != s.selfShardID {
+		return
+	}
+
+	mapBlockTypesTxs := make(map[block.Type]map[string]data.TransactionHandler)
+	for i := 0; i < len(body.MiniBlocks); i++ {
+		miniBlock := body.MiniBlocks[i]
+		if _, ok := mapBlockTypesTxs[miniBlock.Type]; !ok {
+			mapBlockTypesTxs[miniBlock.Type] = s.txCoordinator.GetAllCurrentUsedTxs(miniBlock.Type)
+		}
+
+		marshalizedMiniBlock, errNotCritical := s.marshalizer.Marshal(miniBlock)
+		if errNotCritical != nil {
+			log.Warn("saveAllTransactionsToStorageIfSelfShard.Marshal", "error", errNotCritical.Error())
+			continue
+		}
+
+		errNotCritical = s.storage.Put(dataRetriever.MiniBlockUnit, shardHdr.MiniBlockHeaders[i].Hash, marshalizedMiniBlock)
+		if errNotCritical != nil {
+			log.Warn("saveAllTransactionsToStorageIfSelfShard.Put -> MiniBlockUnit", "error", errNotCritical.Error())
+		}
+	}
+
+	mapTxs := s.importHandler.GetTransactions()
+	// save transactions from imported map
+	for _, miniBlock := range body.MiniBlocks {
+		if miniBlock.SenderShardID == s.selfShardID {
+			continue
+		}
+
+		for _, txHash := range miniBlock.TxHashes {
+			tx, ok := mapTxs[string(txHash)]
+			if !ok {
+				log.Warn("miniblock destination me in genesis block should contain only imported txs")
+				continue
+			}
+
+			unitType := dataRetriever.TransactionUnit
+			switch miniBlock.Type {
+			case block.TxBlock:
+				unitType = dataRetriever.TransactionUnit
+			case block.RewardsBlock:
+				unitType = dataRetriever.RewardTransactionUnit
+			case block.SmartContractResultBlock:
+				unitType = dataRetriever.UnsignedTransactionUnit
+			}
+
+			marshalledData, errNotCritical := s.marshalizer.Marshal(tx)
+			if errNotCritical != nil {
+				log.Warn("saveAllTransactionsToStorageIfSelfShard.Marshal", "error", errNotCritical.Error())
+				continue
+			}
+
+			errNotCritical = s.storage.Put(unitType, txHash, marshalledData)
+			if errNotCritical != nil {
+				log.Warn("saveAllTransactionsToStorageIfSelfShard.Put -> Transaction", "error", errNotCritical.Error())
+			}
+		}
+	}
+}
+
+func (s *shardBlockCreator) saveAllCreatedDSTMeTransactionsToCache(
+	shardHdr *block.Header,
+	body *block.Body,
+) {
+	// no need to save from me, only from other genesis blocks which has results towards me
+	if shardHdr.GetShardID() == s.selfShardID {
+		return
+	}
+
+	mapBlockTypesTxs := make(map[block.Type]map[string]data.TransactionHandler)
+	crossMiniBlocksToMe := make([]*block.MiniBlock, 0)
+	for i := 0; i < len(body.MiniBlocks); i++ {
+		miniBlock := body.MiniBlocks[i]
+		isCrossShardDstMe := miniBlock.SenderShardID == s.shardCoordinator.SelfId() &&
```

Comment:
miniBlock.SenderShardID != s.shardCoordinator.SelfId()

Reply:
this is correct. As I am interested in miniblocks from postProcessor created in this genesis block which are towards my shard.

Reply:
Ok, but the naming is misleading because it is expected as the sender to be other shard but not self